extensions/wikia/MercuryApi/MercuryApiController.class.php

```diff
@@ -347,8 +347,17 @@ public function getPage() {
 			$data['ns'] = $title->getNamespace();
 
 			if ( MercuryApiMainPageHandler::shouldGetMainPageData( $isMainPage ) ) {
+
+				// TODO: remove this line after release of XW-2590 (XW-2625)
 				$data['mainPageData'] = MercuryApiMainPageHandler::getMainPageData( $this->mercuryApi );
```

Comment:
I think it's a cost that we could pay for limited time to make cleanup easier and less error prone.